proposals/wac-ucr/index.bs

```diff
@@ -1114,13 +1114,178 @@ and the presentation goes off perfectly.
 Requirements {#requirements}
 ================================================================================
 
-Issue: Populate requirements based on established use cases.
+## Functional Requirements ## {#req-functional}
+
+1. The system shall permit or restrict an [=agent's=] ability to access a 
+    [=resource=] or [=collection=]. [[#uc-basic]], 
+    [[#uc-collections]], [[#uc-blockagents]]
+
+1. The system shall allow access to be limited based on the [=identity=] of 
+    the [=agent=]. [[#uc-basic]], [[#uc-collections]]
+
+1. The system shall allow access to be limited to an [=agent=] 
+    based on the [=agent's=] membership in a certain group of [=agents=]. 
+    [[#basic-group]]
+    
+1. The system shall allow access to be limited to an [=agent=] 
+    based on the client [=application=] in use by the [=agent=]. 
+    [[#uc-trusted-applications]]
+
+1. The system shall allow for access to a certain resource to be limited to 
+    the [=agent=] that created that resource.
+    [[#collection-readappendwrite]]
+
+1. The system shall allow access to be limited to an [=agent=] 
+    based on the [=agent's=] possession of a certain verifiable credential
+    or capability. FUTURE. 
+    [[#capabilities-vc]], [[#capabilities-link]], [[#uc-minimalcredentials]]
+    
+1. The system shall allow access to be limited to any 
+    unauthenticated [=agent=].
+    [[#basic-public]]
+    
+1. The system shall allow access to be limited to any 
+    authenticated [=agent=].
+    [[#basic-authenticated]]
+
+1. The system shall limit the ability to read the access permissions 
+    associated with a certain [=resource=].
+    [[#basic-control]], [[#uc-whopermitted]], [[#uc-historyofchanges]]
+    
+1. The system shall limit the ability to change the access permissions 
+    associated with a certain [=resource=].
+    [[#basic-control]]
+
+1. The system shall furnish the effective access permission modes on a certain
+    [=resource=] as they relate to the [=agent=] making the request,
+    in the request response.
+    [[#uc-client-determine-access-privileges]]
```

Comment:
Could we use something other than "furnish"? :) Simpler/International English =)

The requirement is perhaps too general. The finer details may be lost - doesn't capture what the UC mentions re distinguishing user, public, group.. Looking at the current requirement description, it asks for an extensible solution, in contrast, the UC is specific.

To be clear, I'm not talking about the preferred solution right now, but what the requirement entails.

Reply:
> Could we use something other than "furnish"?

Changed to provide in https://github.com/solid/authorization-panel/pull/152/commits/b9f4c3de2556f09e1bddc9d1bdfd5b63c942ef89

> The requirement is perhaps too general. The finer details may be lost - doesn't capture what the UC mentions re distinguishing user, public, group

`Effective access permissions as they relate to the agent making the request` is meant to convey that the agent would be provided only the modes of access that apply to them based on the current ruleset. Re-reading the use case, I'm not sure what else you'd be looking to include here?

Reply:
The agent is provided with access modes that are granted to different permission groups, not only the ones that apply to them.

Reply:
> The agent is provided with access modes that are granted to different permission groups, not only the ones that apply to them.

Can you clarify what would fall into these different permission groups that don't apply to the agent? If the agent is provided with information about permissions that don't apply to them, we're exposing authorization data that is meant to be private, unless the agent has been explicitly given access to read all permissions (e.g. control). I could see public as being a special case given that they can also enumerate available public access on their own with no additional knowledge.

Reply:
To paraphrase, I meant permissions besides the ones that are specifically granted to an identifiable agent. The permissions that are made visible are only the ones that the agent is authorized to know. So, for example, the user level permission would be what they are specifically granted, the public permission would be everyone and that includes the user, or a particular group permission in which the user is a member of.

Reply:
Slightly expanded the requirement in https://github.com/solid/authorization-panel/pull/152/commits/7ad7ccb951e8444ad8618465f0bfb5424091ebfa. The requirement doesn't rule out what you're suggesting, but the specifics of how it works should be figured out during the spec work, because there are some risk / security considerations that will need to be evaluated.